test-e2e/test/helpers.js

```diff
@@ -120,12 +120,14 @@ async function createTestSandbox(...fixture: Fixture): Promise<TestSandbox> {
     }
     if (!options.noEsyPrefix) {
       env = {
-        ...env,
         ESY__PREFIX: esyPrefixPath,
         ESYI__CACHE: path.join(esyPrefixPath, 'esyi'),
         ESYI__OPAM_REPOSITORY: `:${opamRegistry.registryPath}`,
         ESYI__OPAM_OVERRIDE: `:${opamRegistry.overridePath}`,
         NPM_CONFIG_REGISTRY: npmRegistry.serverUrl,
+        // The order here is important... the env variable limit can be exhausted,
+        // so we want to apply our overrides first!
+        ...env,
```

Comment:
I don't understand this change — why it help with env variable limit? Also `env` includes current process's env and that means if tests are executed in an environment which has `ESY__PREFIX` (for example) then it will be used instead of the one crafted specifically for a test case.

Reply:
Without this change, on my environment, I'd hit failures like this: ```
  ● Basic tests › it should correctly install a single dependency that contains no sub-dependencies

    Error

      Error: Command failed: E:\esy\_build\default\esy\bin\esyCommand.exe install
      esy: error, exiting...
           no npm package no-deps found
             resolving no-deps@=1.0.0
```

I added some logging in `esyCommand` and found that these environment variables weren't making it in - presumably due to hitting the environment variable limit.

> Also env includes current process's env and that means if tests are executed in an environment which has ESY__PREFIX (for example) then it will be used instead of the one crafted specifically for a test case.

Good point. A couple options:
- 1) Could we just bring in the `PATH` variable and leave the rest?
- 2) If 1 isn't an option, could we delete those overrides off the existing environment?

I think 1) might be a good option because it would keep our environment fairly insulated - I'll test it out and see if tests pass.

Reply:
Interesting... I investigated this more (only bringing in the `PATH` variable isn't sufficient, since we have test cases that pass environment variables to verify they make it through).

This only reproduces for me when I run the tests via `npm run test:e2e` - it seems that npm injects a ton of context into the environment, which causes problems in this case. If I run directly via `jest --runInBand test-e2e`, this problem no longer manifests.

So I'll revert this change and update the appveyor script to use `jest` directly instead of the `npm` command.